odoo/addons/base/models/ir_module.py

```diff
@@ -316,9 +317,11 @@ def _check_external_dependencies(terp):
             return
         for pydep in depends.get('python', []):
             try:
+                # this is for backwards compatibility, to be removed in the future
                 importlib.import_module(pydep)
-            except ImportError:
-                raise ImportError('No module named %s' % (pydep,))
+            except ModuleNotFoundError:
+                # this raises pkg_resources.DistributionNotFound in case of missing dependency
```

Comment:
if the version is not correct, the error is a `pkg_resources.VersionConflict`. It is probably worth catching it to adapt the error message. Otherwise, if I don't have the right version, my error will be
>  Unable to install module "auth_ldap" because an external dependency is not met: python-ldap 3.1.0 

while 3.1 is the version installed on my machine, not the expected target